### PR TITLE
Panic when borrowing would overflow.

### LINKED ISSUE #1: Unsound atomic borrowing
This crate implements atomic borrowing but implementation is unsound.
For example `CellRef` rampant cloning can overflow counter back to `0`

### PR DESCRIPTION
**Pull request by azriel91**
Fixes #1.

### CODE REVIEW

File: src/cell_ref.rs [hunk 1]
```diff
@@ -20,6 +22,22 @@ impl<'a, T> CellRef<'a, T>
 where
     T: ?Sized,
 {
+    /// Returns a clone of this `CellRef`.
+    ///
+    /// This should be used instead of `.clone()` to handle potential overflow
+    /// of references.
+    pub fn try_clone(&self) -> Result<Self, RefOverflow> {
+        let previous_value = self.flag.fetch_add(1, Ordering::Release);
```
Comment (zakarumych) [inline src/cell_ref.rs:30]: I'm sorry but overflow still occurs here.

Comment (azriel91) [reply src/cell_ref.rs:30]: how so? the control flow returns err, so it should be detected right

or, it should `load` first, then only add if the loaded value is not `usize::MAX`

Comment (zakarumych) [reply src/cell_ref.rs:30]: Control flow returns err, and counter is `0`. Now I can borrow thing mutably.

Comment (zakarumych) [reply src/cell_ref.rs:30]: Control flow returns err, and counter is `0`. Now I can borrow thing mutably.

Yes, you could load first, but I suggest to optimize for common flow, e.g. when counter never reaches `usize::MAX`

**Pull request merged**